includes/DataDumpPager.php

```diff
@@ -342,25 +362,22 @@ public function onGenerate( array $params ) {
 				)
 			);
 		}
-
-		return true;
 	}
 
-	private function getGenerateLimit( string $type ) {
-		$config = $this->config->get( 'DataDump' );
+	private function getGenerateLimit( string $type ): bool {
+		$config = $this->config->get( ConfigNames::DataDump );
 
 		if ( isset( $config[$type]['limit'] ) && $config[$type]['limit'] ) {
-			$res = $this->mDb->select(
-				'data_dump',
-				'*',
-				[
-					'dumps_type' => $type
-				]
-			);
+			$res = $this->getDatabase()->newSelectQueryBuilder()
+				->select( '*' )
+				->from( 'data_dump' )
+				->where( [ 'dumps_type' => $type ] )
+				->caller( __METHOD__ )
+				->fetchResultSet();
```

Comment:
[`SelectQueryBuilder::fetchRowCount()`](https://doc.wikimedia.org/mediawiki-core/REL1_42/php/classWikimedia_1_1Rdbms_1_1SelectQueryBuilder.html#ad2f7767fae0bb5bec8482ec224246677)

Probably should try the COUNT() expression example in the documention of the function linked above, but no worries if it doesn't work.

Reply:
Good idea. Done.